cleanlab/classification.py

```diff
@@ -379,17 +433,20 @@ class 0, 1, ..., K-1. `pred_probs` should have been computed using 3 (or
                     "Assigning sample weights for final training based on estimated label quality."
                 )
 
-            self.sample_weight = np.ones(np.shape(labels_cleaned))
+            sample_weight = np.ones(np.shape(labels_cleaned))
             for k in range(self.num_classes):
                 sample_weight_k = 1.0 / max(self.noise_matrix[k][k], 1e-3)  # clip sample weights
-                self.sample_weight[labels_cleaned == k] = sample_weight_k
+                sample_weight[labels_cleaned == k] = sample_weight_k
 
+            sample_weight_expanded = np.zeros(len(labels))  # pad pruned examples with zero weight
```

Comment:
i would move this logic into an accessor for `get_label_issues_df` (see comment below). You could just return self.clf here as is the normal convention and you can build up a dataframe as its own accessor method, you can still store predictions, given labels, label quality scores, and boolean mask in a `self.label_issues_df` (if you don't mind bloating the size of the instance), but i would move all other logic and columns into the accessor method and just add columns for `self.sample_weight`, etc.

Reply:
updated. also added new `self.save_space()` method to delete all the non-sklearn attributes to streamline deployment of the fitted Estimator.

Reply:
nice. i like it!